#543 - Validate id_token signature: Refresh jwks keys from server only if it is allowed

duttarnab opened the pull request:
https://github.com/GluuFederation/oxd/issues/543

yuriyz reviewed (changes requested):
It makes sense to introduce period value, e.g. do not re-fetch 24h. Otherwise if admin sets `allowIdTokenValidationWithRefreshedOpJwks=false` oxd can stuck with outdated keys (depending on keys re-generation interval on server side).

duttarnab commented:
1. We have removed `allowIdTokenValidationWithRefreshedOpJwks=false`
2. The keys are stored in cache for a time-period (time-period is configured in oxd-server.yml file)
https://github.com/GluuFederation/oxd/blob/master/oxd-server/src/main/java/org/gluu/oxd/server/service/PublicOpKeyService.java#L42
3. The oxd read the jwks from OP server (if not present in cache) when the Validator.java is instantiated. 
https://github.com/GluuFederation/oxd/blob/master/oxd-server/src/main/java/org/gluu/oxd/server/op/Validator.java#L79
4. And Validator.java is not singleton (not prototype) so step#3 is repeated each time it is instantiated.

So I think re-fetching of jwks from OP server is a redundant step during id_token validation. Hence I have removed this.
